src/app/qgsvectorlayersavestyledialog.cpp

```diff
@@ -0,0 +1,196 @@
+  connect( mStyleTypeComboBox, qgis::overload<int>::of( &QComboBox::currentIndexChanged ), this, [ = ]( int idx )
+  {
+    Q_UNUSED( idx );
+    QgsVectorLayerProperties::StyleType type = currentStyleType();
+    mSaveToFileWidget->setVisible( type != QgsVectorLayerProperties::DB );
+    mSaveToDbWidget->setVisible( type == QgsVectorLayerProperties::DB );
+    mStyleCategoriesListWidget->setEnabled( type == QgsVectorLayerProperties::QML );
+    mFileWidget->setFilter( type == QgsVectorLayerProperties::QML ? tr( "QGIS Layer Style File (*.qml)" ) : tr( "SLD File (*.sld)" ) );
+  } );
+  mStyleTypeComboBox->addItem( tr( "as QGIS QML style file" ), QgsVectorLayerProperties::QML );
+  mStyleTypeComboBox->addItem( tr( "as SLD style file" ), QgsVectorLayerProperties::SLD );
+  if ( mLayer->dataProvider()->isSaveAndLoadStyleToDatabaseSupported() )
+    mStyleTypeComboBox->addItem( tr( "in database (%1)" ).arg( providerName ), QgsVectorLayerProperties::DB );
+
+
+  // Save to DB setup
+  mDbStyleDescriptionEdit->setTabChangesFocus( true );
+  setTabOrder( mDbStyleNameEdit, mDbStyleDescriptionEdit );
+  setTabOrder( mDbStyleDescriptionEdit, mDbStyleUseAsDefault );
+  mDbStyleUIFileWidget->setDefaultRoot( myLastUsedDir );
+  mDbStyleUIFileWidget->setFilter( tr( "Qt Designer UI file (*.ui)" ) );
+  connect( mDbStyleUIFileWidget, &QgsFileWidget::fileChanged, this, &QgsVectorLayerSaveStyleDialog::readUiFileContent );
+  connect( buttonBox, &QDialogButtonBox::helpRequested, this, &QgsVectorLayerSaveStyleDialog::showDbHelp );
+
+  // save to file setup
+  mFileWidget->setStorageMode( QgsFileWidget::SaveFile );
+  mFileWidget->setDefaultRoot( myLastUsedDir );
+
+  // fill style categories
+  QgsMapLayer::StyleCategories lastStyleCategories = settings.flagValue( QStringLiteral( "style/lastStyleCategories" ), QgsMapLayer::AllStyleCategories );
+  for ( QgsMapLayer::StyleCategory category : qgsEnumMap<QgsMapLayer::StyleCategory>().keys() )
+  {
+    if ( category == QgsMapLayer::AllStyleCategories )
+      continue;
+
+    QgsMapLayer::ReadableStyleCategory readableCategory = QgsMapLayer::readableStyleCategory( category );
+
+    QListWidgetItem *item = new QListWidgetItem( readableCategory.icon(), readableCategory.name(), mStyleCategoriesListWidget );
+    item->setFlags( item->flags() | Qt::ItemIsUserCheckable );
+    item->setCheckState( lastStyleCategories.testFlag( category ) ? Qt::Checked : Qt::Unchecked );
+    item->setData( Qt::UserRole, category );
+  }
+
+  restoreGeometry( settings.value( QStringLiteral( "Windows/saveVectorLayerStyleTo/geometry" ) ).toByteArray() );
+}
+
+void QgsVectorLayerSaveStyleDialog::accept()
+{
+  bool ok = true;
+  QgsSettings settings;
+
+  switch ( currentStyleType() )
+  {
+    case QgsVectorLayerProperties::QML:
+    case QgsVectorLayerProperties::SLD:
+    {
+      if ( mFileWidget->filePath().isEmpty() )
+      {
+        QMessageBox::information( this, tr( "Save Style in file" ), tr( "Select a file destination." ) );
```

Comment:
"Save Style in File"

Reply:
(sorry... just getting in before @DelazJ ;) )

Reply:
:smiley: